src/scriptdev2/scripts/kalimdor/durotar.cpp

```diff
@@ -127,9 +127,10 @@ struct npc_lazy_peonAI : public ScriptedAI
                 }
                 case 2: //spawn
                 {
-                    m_creature->SetFacingTo(m_fSpawnO);
-                    m_creature->GetMotionMaster()->MoveTargetedHome(); //hacky way of getting the peon to sleep
+                    m_creature->GetMotionMaster()->MoveIdle();
                     m_creature->HandleEmote(EMOTE_STATE_NONE);
+                    m_creature->SetFacingTo(m_fSpawnO);
+                    m_creature->CastSpell((Unit*)m_creature, SPELL_PEON_SLEEP, false);
```

Comment:
That's what I did at first, but it didn't want to accept the arguments so I added more and did a cast. Tried again this way now and it works. Weird, maybe my VS was being dumb last time.